### ISSUE TITLE
ProvisionException hides PlayException

### ISSUE THREAD

**Issue by mkurz** [playframework/playframework#10710]
Let's say we have an invalid message file (`conf/messages`):
```
hi=hello
this_line_will_fail_because_there_is_no_value
by=goodbye
```
As you can see the second line will cause a problem.
When Play parses a message file and detects it is invalid, it will throw a `PlayException` which references the line in the invalid file:
https://github.com/playframework/playframework/blob/c395734fa7c838f552843a458e78d0ee9cb805ac/core/play/src/main/scala/play/api/i18n/Messages.scala#L157-L170
This _should_ actually display a nice error page in the browser.

However... when using Guice, what the user sees is a Guice `ProvisionException`:

![108183764-b7ab4a80-710a-11eb-8c9f-2ceb52be902c_new](https://user-images.githubusercontent.com/644927/108337186-22c15380-71d5-11eb-89a4-1b469582a0df.png)

That happens because the `PlayException` gets thrown during initializing of the message module... When Guice can't initialize, it will wrap any exception inside a `ProvisionException` and throws that one instead... I mean look at the `ProvisionException`,
it even says in its own exception message: "_Unable to provision, see the following errors:_"

IMHO a `PlayException` should not be hidden by Guice, so we get our nice error pages, in the above case that would be:

![image](https://user-images.githubusercontent.com/644927/108185621-ca268380-710c-11eb-8ac9-a420fe89f4c6.png)

I will provide a pull request with a fix...

**Comment by mkurz** [comment]
Actually we can enhance the situation even more:
If a component can not be initialized by Play, e.g. because an exception gets thrown in a component's constructor, what you currently see is:
![image](https://user-images.githubusercontent.com/644927/108337448-6916b280-71d5-11eb-9b22-9d251d29c22c.png)
So that again is the Guice `ProvisionException`...

But what we can do is to unwrap the `RuntimeException` + put it inside a `PlayException` so it shows us the source of the problem:
![image](https://user-images.githubusercontent.com/644927/108337635-9e230500-71d5-11eb-80fc-21d202449abb.png)

(fyi: I injected the `SomeComponent` into a controller)

This is much much nicer.
Pull request is coming soon.

**Comment by mkurz** [comment]
Will be fixed by #10711
